providers/monitor.rb

```diff
@@ -25,11 +25,6 @@ def whyrun_supported?
     )
     cookbook new_resource.cookbook
     sensitive true if Chef::Resource.instance_methods(false).include?(:sensitive)
-    notifies :restart, 'service[datadog-agent]', :delayed if node['datadog']['agent_start']
```

Comment:
Any reason for removing the condition `if node['datadog']['agent_start']` from the `notifies` properties that you added in the recipes?